models/utilities/default_model_mod.f90

```diff
@@ -49,8 +51,24 @@ module default_model_mod
           convert_vertical_obs,   &
           convert_vertical_state, &
           read_model_time,        &  ! from the dart_time_io module
-          write_model_time
-
+          write_model_time,       &
+          parse_variables,        &
+          parse_variables_clamp,  &
+          state_var_type,         &
+          MAX_STATE_VARIABLE_FIELDS, &
+          MAX_STATE_VARIABLE_FIELDS_CLAMP
+
+type :: state_var_type
+    integer                        :: nvars
+    character(len=64), allocatable :: netcdf_var_names(:)
```

Comment:
the max length of a netcdf variable name is a constant set by the netcdf library

NF90_MAX_NAME

Its 256 at the moment:
https://github.com/Unidata/netcdf-fortran/blob/6302daa1376f076add3d97794195d070ab81fdd9/fortran/netcdf_constants.F90#L78
```suggestion
    character(len=NF90_MAX_NAME), allocatable :: netcdf_var_names(:)
```